### PR TITLE
Simplify checkpoint connector loading after Checkpoint IO plugin's introduction

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

- Moves the validating logic of checking for the path's existence from the checkpoint connector to the torch checkpoint IO plugin (where the fsspec logic lives)
- Moves the CUDA cache clearing from the checkpoint connector to the training type plugin 
- Renames `load_checkpoint_file` in the training type to `load_checkpoint` for consistency with the checkpoint IO's API

### Does your PR introduce any breaking changes? If yes, please list them.

This is a breaking change for any custom plugin that extended `load_checkpoint_file` as this has been renamed to `load_checkpoint` with this PR

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/plugins/test_checkpoint_io_plugin.py [hunk 1]
```diff
@@ -28,14 +28,14 @@
 
 class CustomCheckpointIO(CheckpointIO):
     save_checkpoint_called: bool = False
-    load_checkpoint_file_called: bool = False
+    load_checkpoint_called: bool = False
```
Comment (awaelchli) [inline tests/plugins/test_checkpoint_io_plugin.py:31]: can be removed, unused
the test below uses a mock
forgotten in the original PR :)

File: pytorch_lightning/plugins/training_type/training_type_plugin.py [hunk 2]
```diff
@@ -154,7 +154,9 @@ def results(self) -> Optional[Union[_EVALUATE_OUTPUT, _PREDICT_OUTPUT]]:
         """
         return self._results
 
-    def load_checkpoint_file(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
+    def load_checkpoint(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
+        if self.on_gpu:
+            torch.cuda.empty_cache()
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/training_type_plugin.py:159]:
```suggestion
        torch.cuda.empty_cache()
```
not needed. call to `empty_cache` is a no-op if cuda is not available

**Review by awaelchli — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Neat !

File: tests/plugins/test_checkpoint_io_plugin.py (hunk 1 above)
Comment (tchaton) [reply tests/plugins/test_checkpoint_io_plugin.py:31]: @ananthsub
